journal/journal.go

```diff
@@ -77,34 +77,24 @@ func Send(message string, priority Priority, vars map[string]string) error {
 		return journalError("could not connect to journald socket")
 	}
 
+	socketAddr, err := net.ResolveUnixAddr("unixgram", "/run/systemd/journal/socket")
+	if err != nil {
+		return journalError(err.Error())
+	}
 	data := new(bytes.Buffer)
 	appendVariable(data, "PRIORITY", strconv.Itoa(int(priority)))
 	appendVariable(data, "MESSAGE", message)
 	for k, v := range vars {
 		appendVariable(data, k, v)
 	}
 
-	_, err := io.Copy(conn, data)
-	if err != nil && isSocketSpaceError(err) {
-		file, err := tempFd()
-		if err != nil {
-			return journalError(err.Error())
-		}
-		defer file.Close()
-		_, err = io.Copy(file, data)
-		if err != nil {
-			return journalError(err.Error())
-		}
-
-		rights := syscall.UnixRights(int(file.Fd()))
-
-		/* this connection should always be a UnixConn, but better safe than sorry */
-		unixConn, ok := conn.(*net.UnixConn)
-		if !ok {
-			return journalError("can't send file through non-Unix connection")
-		}
-		unixConn.WriteMsgUnix([]byte{}, rights, nil)
-	} else if err != nil {
+	/* this connection should always be a UnixConn, but better safe than sorry */
+	unixConn, ok := conn.(*net.UnixConn)
+	if !ok {
+		return journalError("can't send file through non-Unix connection")
+	}
+	_, _, err = unixConn.WriteMsgUnix(data.Bytes(), nil, socketAddr)
```

Comment:
From what I observe, it didn't work.
Even if we switch to a connection-less socket, it seems there is an incompatibility between golang and journald, the message is rejected with _"Got too many file descriptors via native socket. Ignoring."_
I didn't spend a lot of time on it but it looks to come from a condition in journald:
https://github.com/systemd/systemd/blob/master/src/journal/journald-server.c#L1389
where _n_ is supposed to be equal to _0_, but the SendmsgN method in golang seems to always send a dummy byte when the message is empty:
https://github.com/golang/sys/blob/master/unix/syscall_linux.go#L681